src/pages/RFOX/components/Stake/StakeInput.tsx

```diff
@@ -279,32 +302,32 @@ export const StakeInput: React.FC<StakeInputProps & StakeRouteProps> = ({
       functionName: 'approve',
       args: [
         RFOX_PROXY_CONTRACT_ADDRESS,
-        BigInt(toBaseUnit(amountCryptoPrecision, stakingAsset?.precision ?? 0)),
+        BigInt(toBaseUnit(amountCryptoPrecision, asset?.precision ?? 0)),
```

Comment:
paranoid should handle missing asset instead of risk sending precision amount to contract

Reply:
@woodenfurniture that's actually correct, we're approving (and estimating approval fees for) the *staking* asset, not the currently selected asset.

e.g you select FOX.ETH, your (selected) asset is ETH, but staking asset is FOX.ARB.

Regardless of whether you're going to be bridging or not (briding is its own step), the approval fees are the approval fees of *staking* i.e the approval fees of FOX.ARB.

Rubber-ducking myself here, it looks lke naming could be improved to make this clearer? Renamed `assetId` and `asset` to `selectedAssetId` and `selectedAsset` in 1e9985214c

Reply:
@woodenfurniture three birds one stone in 62532f132f

1. Renamed all related `asset` vars to `selectedAsset` for clarity
2. Sane naming makes everything saner, self-reviewed the diff after those renames and noticed a few occurences of selected asset being used in place of staking asset and vice-versa
3. Added paranoia early returns for `!asset`